File: Java/src/main/java/minimal/model/Pot.java
package minimal.model;

import java.util.ArrayList;
import java.util.Collection;
import java.util.List;

public class Pot {

    private final List<Ingredient> ingredients = new ArrayList<>();

    public boolean add(Ingredient e) {
        return ingredients.add(e);
    }

    public boolean addAll(Collection<Ingredient> c) {
        return ingredients.addAll(c);
    }

    public Meal cook() {
        System.out.println("cooking " + ingredients + " - stir stir stir");
        return new Meal();
    }

}

File: TypeScript/src/model/Pot.ts
import { Ingredient } from "./Ingredient";
import { Meal } from "./Meal";

export class Pot {
    
    private readonly ingredients: Ingredient[] = [];

    public add(e: Ingredient): void {
        this.ingredients.push(e);
    }

    public addAll(c: Ingredient[]): void {
        this.ingredients.push(...c);
    }

    public cook(): Meal {
        console.log(`cooking ${this.ingredients} - stir stir stir`);
        return new Meal();
    }
}
